src/cookbook/games/firestore-multiplayer.md

```diff
@@ -0,0 +1,469 @@
+in between player interactions on local machines.
+Therefore, they don't need to synchronize game states that often.
+
+![An illustration of two mobile phones and a two-way arrow between them](/assets/images/docs/cookbook/multiplayer-two-mobiles.jpg){:.site-illustration}
+
+If you can choose low tick rates as a developer, you should.
+Low tick lowers latency requirements and server costs.
+Sometimes, a game requires high tick rates of synchronization.
+For those cases, solutions such as Firestore *don't make a good fit*.
+Pick a dedicated multiplayer server solution such as
+[Nakama](https://heroiclabs.com/nakama/).
+Nakama has a [Dart package](https://pub.dev/packages/nakama).
+
+If you expect that your game requires a low tick rate of synchronization,
+continue reading.
+
+This recipe demonstrates how to use the
+[`cloud_firestore` package](https://pub.dev/packages/cloud_firestore)
+to implement multiplayer capabilities in your game.
+This recipe doesn't require a server.
+It uses two or more clients sharing game state via Cloud Firestore.
+
+
+## 1. Prepare your game for multiplayer
+
+Write your game code to allow changing the game state in response to both
+local events and remote events.
+A local event could be a player action or some game logic.
+A remote event could be a world update coming from the server.
+
+![Screenshot of the card game](/assets/images/docs/cookbook/multiplayer-card-game.jpg){:.site-mobile-screenshot .site-illustration}
+
+To simplify this cookbook recipe, start with
+the `card` template that you'll find
+in the [`flutter/games` repository](https://github.com/flutter/games).
+Run the following command to clone that repository:
+
+```shell
+$ git clone https://github.com/flutter/games.git
+```
+
+{% comment %}
+  If/when we have a "sample_extractor" tool, or any other easier way
+  to get the code, mention that here.
+{% endcomment %}
+
+Open the project in `templates/card`.
+
+{{ site.alert.note }}
+  You can ignore this step and follow the recipe with your own game
+  project. Adapt the code at appropriate places.
+{{ site.alert.end }}
+
+
+## 2. Install Firestore
+
+[Cloud Firestore](https://cloud.google.com/firestore/)
+is a horizontally scaling NoSQL document database in the cloud.
+It includes built-in live synchronization.
+This is perfect for our needs.
+It keeps the game state updated in the cloud database,
+so every player sees the same state.
+
+If you want a quick, 15-minute primer on Cloud Firestore,
+watch the following video:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/v_hR4K4auoQ" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+To add Firestore to your Flutter project,
+follow the first two steps of the
+[Get started with Cloud Firestore](https://firebase.google.com/docs/firestore/quickstart)
+guide.
```

Comment:
I'm a bit worried that "first two steps" will get out of date and as a reader, it may not be clear if this means the numbered steps within "Create a Cloud Firestore database" or both "Create a Cloud Firestore database" and "Set up your development environment".

Could we perhaps be more explicit somehow? If we directly linked to the related headers, we could catch changes automatically with the linkchecker.

```suggestion
To add Firestore to your Flutter project,
follow the first two steps of the
[Get started with Cloud Firestore](https://firebase.google.com/docs/firestore/quickstart)
guide:

- [Create a Cloud Firestore database](https://firebase.google.com/docs/firestore/quickstart#create)
- [Set up your development environment](https://firebase.google.com/docs/firestore/quickstart#set_up_your_development_environment)

```